src/utils/serviceApiClient.js

```diff
@@ -21,6 +21,92 @@ const {
 
 logger.setLevel(logger.levels[LOG_LEVEL || 'INFO'])
 
+export const externalAPIRequest = cache(
+  async (request, response, cacheRequest = false) => {
+    const cacheKey = encodeURIComponent(request.url)
+
+    if (cacheRequest && request.cache && request.cache.has(cacheKey)) {
+      const { data } = request.cache.get(cacheKey)
+
+      response.setHeader(
+        'Cache-Control',
+        `public,max-age=${CACHE_MAX_AGE_IN_SECONDS}`
+      )
+      response.setHeader('X-Cache', 'HIT')
+
+      return data
+    }
+
+    const cookies = cookie.parse(request.headers.cookie ?? '')
+    const token = cookies[GSSO_TOKEN_NAME]
+
+    const headers = {
+      'x-api-key': REPAIRS_SERVICE_API_KEY,
+      'x-hackney-user': token,
+      Authorization: token,
+      'Content-Type': 'application/json',
+    }
+
+    let { path, ...queryParams } = request.query
+
+    const api = axios.create()
+
+    // Log request
+    api.interceptors.request.use((request) => {
```

Comment:
Might be worth refactoring this out in to a logging utility method

Reply:
Happy to do refactoring on this file, but I think it would be easier to do in a separate PR. This PR is already quite large